File: build-www.js
/**
 * Penny Buddy — Build script
 * Copies web assets into www/ folder for Capacitor native builds
 */
const fs = require('fs');
const path = require('path');

const SRC = __dirname;
const DEST = path.join(__dirname, 'www');

// Files and folders to copy into www/
const FILES = [
  'index.html',
  'app.js',
  'styles.css',
  'manifest.json',
  'sw.js',
  'case-study.html'
];
const DIRS = ['icons'];

// Clean and recreate www/
if (fs.existsSync(DEST)) {
  fs.rmSync(DEST, { recursive: true });
}
fs.mkdirSync(DEST, { recursive: true });

// Copy files
for (const file of FILES) {
  const src = path.join(SRC, file);
  if (fs.existsSync(src)) {
    fs.copyFileSync(src, path.join(DEST, file));
    console.log(`  Copied ${file}`);
  }
}

// Copy directories
for (const dir of DIRS) {
  const srcDir = path.join(SRC, dir);
  const destDir = path.join(DEST, dir);
  if (fs.existsSync(srcDir)) {
    fs.mkdirSync(destDir, { recursive: true });
    const files = fs.readdirSync(srcDir);
    for (const f of files) {
      const ext = path.extname(f).toLowerCase();
      // Only copy assets, skip scripts
      if (['.png', '.jpg', '.svg', '.ico', '.webp'].includes(ext)) {
        fs.copyFileSync(path.join(srcDir, f), path.join(destDir, f));
        console.log(`  Copied ${dir}/${f}`);
      }
    }
  }
}

console.log('\n  www/ folder ready for Capacitor sync.');

File: build_www.py
"""Build www/ folder for Capacitor native builds."""
import os
import shutil

SRC = os.path.dirname(os.path.abspath(__file__))
DEST = os.path.join(SRC, 'www')

FILES = ['index.html', 'app.js', 'styles.css', 'manifest.json', 'sw.js', 'case-study.html']
ICON_EXTS = {'.png', '.jpg', '.svg', '.ico', '.webp'}

# Clean and recreate www/
if os.path.exists(DEST):
    shutil.rmtree(DEST)
os.makedirs(DEST)

# Copy files
for f in FILES:
    src = os.path.join(SRC, f)
    if os.path.exists(src):
        shutil.copy2(src, os.path.join(DEST, f))
        print(f'  Copied {f}')

# Copy icons
icons_src = os.path.join(SRC, 'icons')
icons_dest = os.path.join(DEST, 'icons')
if os.path.exists(icons_src):
    os.makedirs(icons_dest, exist_ok=True)
    for f in os.listdir(icons_src):
        if os.path.splitext(f)[1].lower() in ICON_EXTS:
            shutil.copy2(os.path.join(icons_src, f), os.path.join(icons_dest, f))
            print(f'  Copied icons/{f}')

print('\n  www/ folder ready for Capacitor sync.')
